endpoints/k8s/k8s

```diff
@@ -661,15 +689,23 @@ function build_pod_spec() {
         echo '          }' >>$json
         echo '        ]' >>$json
     fi
-    if [ "$type" == "cs" -a "$hugepage" == "1" ]; then
-        echo '       ,"volumes": [' >>$json
-        echo '          { "name": "hugepage",' >>$json
-        echo '            "emptyDir": {' >>$json
-        echo '              "medium": "HugePages"' >>$json
-        echo '            }' >>$json
-        echo '          }' >>$json
-        echo '        ]' >>$json
+    # If user provided volumes.json, use it. It not, and "hugepage", do implicitly
+    if [ "$type" == "cs" ]; then
+        set +u
+        if [ ! -z "${volumes[$name]}" ]; then
```

Comment:
again, `! -z` should be replaced with `-n`

Reply:
Ditto aboout preserving same style (using " ! -z") in this function